UnhandledExceptionTelemetryModule.CreateCrashTelemetry should handle AggregateException

schuettecarsten opened the pull request:
As described in #85, when an unhandled exception is caught from Tasks, the exception might not contain a valid information or stack trace data.

So this PR implements the following idea: UnhandledExceptionTelemetryModule.CreateCrashTelemetry will now handle AggregateException, flatten it and then try to extract the inner exception if there is only one. On an unhandled Task exception, AggregateException mostly only contains one exception.

msftclas commented:
Hi __@schuettecarsten__, I'm your friendly neighborhood Microsoft Pull Request Bot (You can call me MSBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <p>The agreement was validated by Microsoft and real humans are currently evaluating your PR.</p>

TTYL, MSBOT;

izikl commented:
@schuettecarsten looks good. We will run few tests before merging this code.

schuettecarsten commented:
I am not sure if it is a good idea to unwrap the AggregateException at this place in the code or if it would be a better idea to unwrap it before any exception details are added to the crash telemetry?

izikl commented:
Until we will release a Nuget with a fix, you can traverse over the AggregateException using the Description callback:
https://support.hockeyapp.net/discussions/problems/58502-uwp-how-to-call-trackexception?mail_type=queue#comment_40175216

schuettecarsten commented:
@izikl I use the description callback, but it's not the same. What do you think of my changes, are they acceptable?

izikl commented:
It looks good, as you mentioned it has a drawback. 

Our team is on vacation for the holidays, but we plan to look at it after the new year. Not sure exactly when. 

Happy new year 🎆

schuettecarsten commented:
I've rebased the PR and moved the code to flatten the AggregateException into a separate method. Also, flattening is now called before any exception details are written into the CrashTelemetry result to make sure that the exception type and description is also unwrapped.

dougnutz commented:
We are looking at this currently. We plan to address it by the end of the week

dougnutz commented:
We are planning to address this by the end of the week.
